sources/api/netdog/src/main.rs

```diff
@@ -44,206 +48,119 @@ lazy_static! {
-    let mut interface_type = None;
-    let mut interface_family = None;
-    let mut data_file = None;
-
-    while let Some(arg) = iter.next() {
-        match arg.as_ref() {
-            "-i" => {
-                let value = iter
-                    .next()
-                    .unwrap_or_else(|| usage_msg("Did not give argument to -i"));
-                interface_name = Some(
-                    serde_plain::from_str::<InterfaceName>(&value)
-                        .unwrap_or_else(|_| usage_msg(format!("Unknown interface name {}", value))),
-                );
-            }
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(subcommand, name = "remove")]
+// `wicked` may call `remove` with the below args and failing to parse them can cause an error in
+// `wicked`.
+/// Not implemented
+struct RemoveArgs {
+    #[argh(option, short = 'i')]
+    /// name of the network interface
+    interface_name: InterfaceName,
 
-            "-t" => {
-                let value = iter
-                    .next()
-                    .unwrap_or_else(|| usage_msg("Did not give argument to -t"));
-                interface_type = Some(
-                    serde_plain::from_str::<InterfaceType>(&value)
-                        .unwrap_or_else(|_| usage_msg(format!("Unknown interface type {}", value))),
-                );
-            }
+    #[argh(option, short = 't')]
+    /// network interface type
+    interface_type: InterfaceType,
 
-            "-f" => {
-                let value = iter
-                    .next()
-                    .unwrap_or_else(|| usage_msg("Did not give argument to -f"));
-                interface_family = Some(
-                    serde_plain::from_str::<InterfaceFamily>(&value).unwrap_or_else(|_| {
-                        usage_msg(format!("Unknown interface family {}", value))
-                    }),
-                );
-            }
+    #[argh(option, short = 'f')]
+    /// network interface family (ipv4/6)
+    interface_family: InterfaceFamily,
 
-            // `wicked` may call this with "info" as the final argument, so if
-            // we already have a data file then we're done.
-            p => match data_file {
-                None => data_file = Some(PathBuf::from(p)),
-                Some(_) => break,
-            },
-        }
-    }
+    #[argh(positional)]
+    /// lease info data file
+    data_file: PathBuf,
 
-    Ok((
-        sub_command,
-        Some(Args {
-            interface_name: interface_name.unwrap_or_else(|| usage()),
-            interface_type: interface_type.unwrap_or_else(|| usage()),
-            interface_family: interface_family.unwrap_or_else(|| usage()),
-            data_file: data_file.unwrap_or_else(|| usage()),
-        }),
-    ))
+    #[argh(positional)]
+    // wicked might add `info` to the call to this program.  We don't do anything with it but must
+    // be able to parse the option to avoid failing
+    /// ignored
+    info: Option<String>,
```

Comment:
Since we previously required these args this was already broken. :fearful: 

With the understanding of the `wicked` code you mentioned, I removed those arguments.  Testing it is going to be really complicated, but I feel pretty comfortable with the change given our understanding of the `wicked` source.